### DISCUSSION TITLE
File extension flexibility

### DISCUSSION THREAD

**Discussion by thejevans** [ransome1/sleek#512] [Ideas]
I've been using sleek for a while now, and I love it. Would it be possible to either allow users to try to load any non-registered file type as if it was a todo.txt file, or to give an option to allow users to register other file extensions as todo.txt files?

## Context
I recently started using Obsidian as a general note taking system, and there is a todo.txt plugin for it that works pretty well, but it doesn't have a lot of the features that sleek has that make todo.txt a really nice todo system.

The advantages of using the Obsidian plugin that makes me want to use it in addition to sleek are that it is by far the best way I've found to interact with todo.txt on my phone, and I share some obsidian vaults with my partner, so it's a great way to share my todo.txt with them.

The disadvantage is that the way the plugin system works for Obsidian is that plugins need to register file extensions that they are designed to work for, and are unable to do any regex on the file names themselves. This led the developer of the plugin to require todo.txt files to instead have .todotxt extensions. They have an experimental option to allow users to register other file extensions, but if I choose to register .txt files, then all .txt files will be rendered as todo.txt files when I open them in Obsidian, which is not ideal.

My current workflow is to make hardlinks on my PCs like this `~/todotxt/todo-<name>.txt` -> `~/obsidian/<vault>/<name>.todotxt`, which works, but is a bit annoying and kind of hacky.

Given that sleek's whole thing is todo.txt files, I don't see a downside to adding this feature other than if it's difficult to implement. Thoughts?

**Comment by ransome1** [comment]
Hey @thejevans thanks for reaching out. I remember that early on in sleek development, we extended sleeks file type compatiblity to .md files in order to make sleek more compatible with Obsidian. And I believe it didn't really hurt anybody.

I think what we could discuss is a setting in sleek, 
- which would allow you to basically add all file types (simply a toggle)
- or, although more work on our side, would be to let the user manually add the required file type extensions him or herself (a text field, in which you could add file type extensions, separated by commas)

Unless somebody tells me, there could some security risk to this.

**Reply by thejevans** [reply]
Either would be awesome!

**Comment by dnngll** [comment]
Hey @thejevans ,

I might be getting this wrong, but wasn't this always possible? Here's a screenshot using 1.3.1:

![2023 09 13, 16 02 48](https://github.com/ransome1/sleek/assets/12801866/cd20d0cc-4418-4a59-9a47-401e073fa91e)

And just because I'm curious and didn't find a proper plugin for Obsidian yet: Are you using https://github.com/mvgrimes/obsidian-todotxt-plugin ?

Thanks

**Reply by thejevans** [reply]
Yes, that is the Obsidian plugin I am using. If you read the documentation for it, you'll find it requires `*.todotxt` files. Sleek does allow other file extensions than `*.txt`, but only `*.md` and `*.todo`, which are hard coded. See #514 for the continued discussion on this topic.

**Reply by dnngll** [reply]
Thanks!

Btw, I was able to open my `! ToDo.txt.md.todotxt` using 1.3.1 without any problems...
